### ISSUE TITLE
Unable to upload app with v4.0.0 added as a dependency

### ISSUE THREAD

**Issue by bdrelling** [Appboy/appboy-ios-sdk#285]
So `AppboyKit/Appboy.bundle/Info.plist` contains keys reference `iossimulator`, which App Store Connect / TestFlight is rejecting. 

```
ERROR [2021-03-15 22:06:12.96]: [Transporter Error Output]: ERROR ITMS-90542: "Invalid CFBundleSupportedPlatforms value. The key 'CFBundleSupportedPlatforms' in the Info.plist file in bundle 'Payload/SpotHero.app/Appboy_iOS_SDK_AppboyKit.bundle/Appboy.bundle' contains an invalid value '[iPhoneSimulator]'. Consider removing the CFBundleSupportedPlatforms key from the Info.plist. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."
DEBUG [2021-03-15 22:06:12.96]: [Transporter]: DBG-X: The error code is: 1102
DEBUG [2021-03-15 22:06:12.96]: [Transporter]: INFO: Done performing authentication.
ERROR [2021-03-15 22:06:13.05]: Transporter transfer failed.
WARN [2021-03-15 22:06:13.05]: 
ERROR [2021-03-15 22:06:13.05]: ERROR ITMS-90542: "Invalid CFBundleSupportedPlatforms value. The key 'CFBundleSupportedPlatforms' in the Info.plist file in bundle 'Payload/SpotHero.app/Appboy_iOS_SDK_AppboyKit.bundle/Appboy.bundle' contains an invalid value '[iPhoneSimulator]'. Consider removing the CFBundleSupportedPlatforms key from the Info.plist. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."
```

And we can see these values here:

<img width="563" alt="image" src="https://user-images.githubusercontent.com/3022693/111369262-dbf43b80-8664-11eb-9496-8ade5c87c29d.png">

* [x] I have checked [Braze's documentation](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/) and am still experiencing the issue.

# Report

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 11+
 SDK Version             | 4.0.0
 Integration Method      | SPM
 Xcode Version           | 12.4
 Repro rate              | 100%
 
## What did you do?
Tried to upload the app to TestFlight.

## What did you expect to happen?
For the app to upload to TestFlight successfully.

## What happened instead?
An error occurred and it didn't upload successfully.

## Steps to reproduce
- Add Appboy iOS SDK v4.0.0 to your project via SPM.
- Try to deploy the app to TestFlight.
- Error occurs.

## Code Snippet
Not a code snippet, but here's the error again:

```
ERROR [2021-03-15 22:06:12.96]: [Transporter Error Output]: ERROR ITMS-90542: "Invalid CFBundleSupportedPlatforms value. The key 'CFBundleSupportedPlatforms' in the Info.plist file in bundle 'Payload/SpotHero.app/Appboy_iOS_SDK_AppboyKit.bundle/Appboy.bundle' contains an invalid value '[iPhoneSimulator]'. Consider removing the CFBundleSupportedPlatforms key from the Info.plist. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."
DEBUG [2021-03-15 22:06:12.96]: [Transporter]: DBG-X: The error code is: 1102
DEBUG [2021-03-15 22:06:12.96]: [Transporter]: INFO: Done performing authentication.
ERROR [2021-03-15 22:06:13.05]: Transporter transfer failed.
WARN [2021-03-15 22:06:13.05]: 
ERROR [2021-03-15 22:06:13.05]: ERROR ITMS-90542: "Invalid CFBundleSupportedPlatforms value. The key 'CFBundleSupportedPlatforms' in the Info.plist file in bundle 'Payload/SpotHero.app/Appboy_iOS_SDK_AppboyKit.bundle/Appboy.bundle' contains an invalid value '[iPhoneSimulator]'. Consider removing the CFBundleSupportedPlatforms key from the Info.plist. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."
```

## Project that demonstrates the issue
Not applicable.

**Comment by chshapiro** [comment]
Hey @bdrelling,

Thanks for reporting! We've identified the issue and are preparing the fix for release.

**Comment by bdrelling** [comment]
@chshapiro this issue has basically halted our development, as we are unable to run the application without a full clean and rebuild every single time, which is unsustainable. When can we expect a fix? Can we get an update?

**Comment by chshapiro** [comment]
Hey @bdrelling,

We're aiming to have this released by tomorrow. We'll update this thread when the release is live.

**Comment by Nonnus** [comment]
following

**Comment by chshapiro** [comment]
Hey all,

We just released version 4.0.1 of the SDK which includes a fix for this issue.

**Comment by bdrelling** [comment]
@chshapiro We're still waiting on updates to the Segment SDK, which is still using version 3.x: https://github.com/Appboy/appboy-segment-ios/blob/master/Package.swift#L16

When will that SDK be updated?

**Comment by chshapiro** [comment]
Hey @bdrelling,

We will release an update to the Segment SDK early next week.

**Comment by chshapiro** [comment]
Hey @bdrelling,

We just released version 4.0.0 of the Segment SDK which uses 4.0.1 of the iOS SDK.

**Comment by bdrelling** [comment]
@chshapiro Thanks. I think this currently can't work for us, though, because we still have the same issue as defined here:

https://github.com/braze-inc/braze-ios-sdk/issues/1

We are currently stuck having to pin to `3.31.1` instead, which requires us to keep a local fork of `appboy-segment-ios` to even be able to pull these dependencies in properly.

**Comment by bdrelling** [comment]
@chshapiro I'm also happy to open up a new issue, I'm just not entirely sure what the process is.

**Comment by chshapiro** [comment]
Hey @bdrelling,

We're looking into your issue and will continue to engage on braze-inc/braze-ios-sdk#1

**Comment by bdrelling** [comment]
Sounds good. Thanks! We'll add additional detail over there as we come across it.
